framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3500,6 +3500,27 @@ protected void checkEnclosingExpr(NewClassTree node, AnnotatedExecutableType con
                 argumentReceiverType = atypeFactory.getAnnotatedType(node.getEnclosingExpression());
             } else {
                 argumentReceiverType = atypeFactory.getReceiverType(node);
+                /*
+                  Consider the following code:
+                  Class Outer {
+                      Class Inner{}
+                  }
+                  Class Top{
+                      void test(Outer outer) {
+                          outer.new Inner(){};
+                      }
+                  }
+                  In Java versions below 11, the argumentReceiverType of outer.new Inner(){} is Top instead of Outer, because java8 organizes
```

Comment:
So is it "below 11" or "java8"?

Reply:
I just notice that I had the same comment again. Why did you just resolve this instead of addressing it??

Reply:
Sorry, I think I wrongly clicked the resolve botton. Usually I address the comment in my local then I click resolve.

Reply:
Done in my local.